File: 5_D/python/Mail.py
from Iletisim import Iletisim
from Kullanici import Kullanici
from abc import ABC,abstractmethod

class Mail(Iletisim):
    
    def bildirimGoster(self,mail,mesaj):
        print(f"{mail} adresine {mesaj} gönderildi")
   
    def bilgiGonder(self,kullanici,mesaj):
        self.bildirimGoster(kullanici.getMail(),mesaj)
     
    
        

File: SOLID/5_D/java/src/Mail.java
public class Mail implements Iletisim {
    
    public void mailGonder(String mail,String mesaj){
        System.out.println(mail+" adresine "+mesaj+" gonderildi");
    }
    public void bilgiGonder(Kullanici kullanici,String mesaj){
        mailGonder(kullanici.getMail(), mesaj);
    }
   
}
